src/components/FileRepoSidebar/ReportsDownloadInput.js

```diff
@@ -36,7 +37,14 @@ export default ({ api, projectId, theme, sqon }) => (
                   },
                 },
                 columns: state.columns,
-              })();
+              })().then(downloadData => {
```

Comment:
would it make sense to move these events into the file download helper functions themselves? that way regardless of where / when they're called in the UI the events still get tracked the same

Reply:
wanted to try and keep these as reasonably close to the display layer as possible, since they are tracking user events more so than programatic events. Moving them to the service layer would require passing contextual information in args which felt like crossing the separation of concerns.

Reply:
fair :)